### PR TITLE
feat(dotenv): allow to restrict env lookup to specific Env variables

### PR DESCRIPTION
**Pull request by mechanoid**
Add `restrictEnvAccessTo` option for `config` and `configSync` to provide an interface corresponding to the `--allow-env=ENV1,ENV2,...` CLI interface. This allows to reduce the exposed Environment of a service to the minimum when using dotenv.

Addressing: #2534

### CODE REVIEW

File: dotenv/mod_test.ts [hunk 1]
```diff
@@ -643,3 +644,55 @@ Deno.test("stringify", async (t) => {
...
+  });
+
+  const decoder = new TextDecoder();
+  const conf = JSON.parse(decoder.decode(stdout).trim());
+
+  assertEquals(conf.GREETING, "hello world", "fetches .env by default");
+  assertEquals(conf.DEFAULT1, "Some Default", "default value loaded");
+});
+
+Deno.test("use of restrictEnvAccessTo for an Env var, without granting read permissions still fails", async () => {
```
Comment (kt3k) [inline dotenv/mod_test.ts:684]:
```suggestion
Deno.test("use of restrictEnvAccessTo for an Env var, without granting env permissions still fails", async () => {
```

**Review by kt3k — APPROVED** [review]
Sorry for the delay in review.

LGTM with nits

File: dotenv/mod_test.ts (hunk 1 above)
Comment (mechanoid) [reply dotenv/mod_test.ts:684]: The delay is very ok, happy that anyone  has time for that at all!

The suggested change is of course welcome 👍

**Comment by mechanoid** [comment]
Ah, you have already pushed the fix for the description! Thank you very much, much appreciated!

**Pull request merged**